### PR TITLE
Feb 26 Merge of Develop Into Master for QA Demo

### PR DESCRIPTION
**Pull request by godfreyyeung**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,6 +31,7 @@
     "@nestjs/core": "^6.0.0",
     "@nestjs/platform-express": "^6.0.0",
     "@nestjs/typeorm": "^6.1.3",
+    "@octokit/rest": "^16.43.1",
```
Comment (allthesignals) [inline package.json:34]: recaptcha fix

File: package.json [hunk 2]
```diff
@@ -43,6 +44,7 @@
     "cookie-parser": "^1.4.4",
     "dotenv": "^8.1.0",
     "eol": "^0.9.1",
+    "express-recaptcha": "^5.0.1",
```
Comment (allthesignals) [inline package.json:47]: recaptcha fix

File: src/project/_utils/build-projects-sql.ts [hunk 3]
```diff
@@ -29,7 +29,7 @@ export function buildProjectsSQL(req, type = 'filter') {
     dcp_femafloodzonecoastala = false,
     dcp_femafloodzonea = false,
     dcp_femafloodzoneshadedx = false,
-    dcp_publicstatus = ['Filed','In Public Review','Completed','Unknown'],
+    dcp_publicstatus = ['Prefiled', 'Filed','In Public Review','Completed','Unknown'],
```
Comment (allthesignals) [inline src/project/_utils/build-projects-sql.ts:32]: recodings of public statuses

**Review by godfreyyeung — COMMENTED** [review]
Looks good to me

File: src/project/project.controller.spec.ts [hunk 4]
```diff
@@ -26,6 +27,10 @@ describe('Project Controller', () => {
           // how you provide the injection token in a test instance
           useValue: new (class ProjectServiceMock { }),
         },
+        {
+          provide: ConfigService,
+          useValue: new (class ConfigServiceMock { }),
+        }
```
Comment (allthesignals) [inline src/project/project.controller.spec.ts:33]: unit test fixes

File: src/project/project.controller.ts [hunk 5]
```diff
@@ -71,4 +76,32 @@ export class ProjectController {
...
+            error,
+          });
+        };
+      } else {
+        response.status(403).send({
+          status: 'captcha invalid',
+        });
+      }
+    });
+  }
```
Comment (allthesignals) [inline src/project/project.controller.ts:106]: recaptcha fix

File: src/project/project.service.ts [hunk 6]
```diff
@@ -82,7 +82,7 @@ export class ProjectService {
      */
     const projectIds = await this.projectRepository.query(buildProjectsSQL(request, 'projectids'));
     const projectIdsString = projectIds.map(d => d.projectid).map(d => `'${d}'`).join(',');
-    const tileSQL = pgp.as.format(tileQuery, { projectIds: projectIdsString });
+    const tileSQL = this.tiles.generateTileSQL(projectIdsString);
```
Comment (allthesignals) [inline src/project/project.service.ts:85]: Related to light refactor for generating tile sql

File: src/project/project.service.ts [hunk 7]
```diff
@@ -186,4 +186,21 @@ export class ProjectService {
...
+      auth: this.config.get('GITHUB_ACCESS_TOKEN'),
+    });
+
+    return await octokit.issues.create({
+      owner: 'nycplanning',
+      repo: 'dcp-zap-data-feedback',
+      title: `Feedback about ${projectname}`,
+      body: `Project ID: [${projectid}](https://zap.planning.nyc.gov/projects/${projectid})\nFeedback: ${text}`,
+    });
+  }
```
Comment (allthesignals) [inline src/project/project.service.ts:205]: recaptcha fix

File: src/project/tiles/tiles.service.ts [hunk 8]
```diff
@@ -10,6 +10,7 @@ import { Project } from '../project.entity';
 import { getQueryFile } from '../../_utils/get-query-file';
 
 const generateVectorTile = getQueryFile('/helpers/generate-vector-tile.sql');
+const tileQuery = getQueryFile('helpers/tile-query.sql');
```
Comment (allthesignals) [inline src/project/tiles/tiles.service.ts:13]: refactor for fix of tile query

File: src/project/tiles/tiles.service.ts [hunk 9]
```diff
@@ -37,9 +38,14 @@ export class TilesService {
...
 
+  generateTileSQL(projectIds: string) {
+    return pgp.as.format(tileQuery, { projectIds });
+  }
+
   async generateTile(type, tileId, x, y, z) {
     // retreive the projectids from the cache
-    const tileQuery = await this.get(tileId);
+    // default to a blank query if no cache is found
+    const tileQuery = await this.get(tileId) || this.generateTileSQL('-1');
```
Comment (allthesignals) [inline src/project/tiles/tiles.service.ts:48]: tile query fix

**Review by allthesignals — APPROVED** [review]

**Pull request merged**